contracts/math/andromeda-graph/src/contract.rs

```diff
@@ -191,14 +218,32 @@ pub fn execute_store_coordinate(
     match allow_negative {
         true => {
             ensure!(
-                x_coordinate >= -(x_length / 2_f64) && x_coordinate <= x_length / 2_f64,
+                x_coordinate.ge(&SignedDecimal::negative_one()
+                    .checked_mul(
+                        x_length
+                            .checked_div(SignedDecimal::percent(200))
+                            .map_err(|_| ContractError::Underflow {})?
+                    )
+                    .map_err(|_| ContractError::Underflow {})?)
+                    && x_coordinate.le(&x_length
+                        .checked_div(SignedDecimal::percent(200))
+                        .map_err(|_| ContractError::Underflow {})?),
```

Comment:
This section is pretty difficult to follow, I'd suggest breaking each element out to a named variable to describe what is occuring.